blocks/record.go

```diff
@@ -0,0 +1,46 @@
+package blocks
+
+import (
+	"io"
+
+	"github.com/bsm/go-sparkey"
+)
+
+// A Record is one key/value pair loaded from a block.
+type Record struct {
+	ValueLen uint64
+
+	iterPool iterPool
+	iter     *sparkey.HashIter
+	io.Reader
+}
+
+func (b *Block) get(key []byte) (*Record, error) {
+	iter, err := b.iterPool.getIter()
+	if err != nil {
+		return nil, err
+	}
+
+	if err := iter.Seek(key); err != nil {
+		return nil, err
+	}
+
+	if iter.State() == sparkey.ITERATOR_ACTIVE {
+		return &Record{
+			ValueLen: iter.ValueLen(),
+			Reader:   iter.ValueReader(),
+			iterPool: b.iterPool,
+			iter:     iter,
+		}, nil
+	}
+
+	// The key doesn't exist, so put the iterator back in the pool.
+	b.iterPool.Put(iter)
+	return nil, nil
+}
+
+func (r *Record) Close() {
```

Comment:
Nothing, but I control the calling code. [`Closer`](https://golang.org/pkg/io/#Closer) says the behavior after the first call is undefined, anyway.